main.py

```diff
@@ -70,42 +71,77 @@
+
+
+    ### LSA ###
+    logger.info("LSA")
 
     # French
-    french_series = sentiment_df[sentiment_df['language'] == 'fr'].iloc[:number_of_articles]['content']
-    #french_series = sentiment_df[sentiment_df['language'] == 'fr']['content']
-    french_model, _, _ = TopicModellingService.fit_lda(texts=french_series, language="french", num_topics=number_of_topics, dataset_passes=ds_passes)
-    french_top_words_per_topic = TopicModellingService.lda_top_words_per_topic(model=french_model, n_top_words=number_of_top_words)
-    #model_path = TopicModellingService.save_gensim_model(model=french_model, language="french")
-    #model1 = TopicModellingService.load_gensim_model(language="french")
-    #model2 = TopicModellingService.load_gensim_model(full_model_path=model_path)
+    french_lsa_model, _, _ = TopicModelingService.fit_model(texts=french_series, language="french", num_topics=number_of_topics, dataset_passes=ds_passes, technique='lsa')
+    df_lsa_topics_fr = TopicModelingService.get_top_words_per_topic(model=french_lsa_model, n_top_words=number_of_top_words, technique='lsa')
+    FileService.df_to_csv(df=df_lsa_topics_fr, file_name="lsa_topics_fr")
+
+    # German
+    german_lsa_model, _, _ = TopicModelingService.fit_model(texts=german_series, language="german", num_topics=number_of_topics, dataset_passes=ds_passes, technique='lsa')
+    df_lsa_topics_de = TopicModelingService.get_top_words_per_topic(model=german_lsa_model, n_top_words=number_of_top_words, technique='lsa')
+    FileService.df_to_csv(df=df_lsa_topics_de, file_name="lsa_topics_de")
+
+    # Matching topics Query: French / Corpus: German
+    best_lsa_matches_by_sentence_transformer_fr_de, _, _, _ = TopicMatcherService.match_by_sentence_transformer(df_lsa_topics_de['Word'], df_lsa_topics_fr['Word'], number_of_top=number_of_top_words, match_score=match_score, print_matches=True)
+
+    # Matching topics Query: German / Corpus: French
+    best_lsa_matches_by_sentence_transformer_de_fr, _, _, _ = TopicMatcherService.match_by_sentence_transformer(df_lsa_topics_fr['Word'], df_lsa_topics_de['Word'], number_of_top=number_of_top_words, match_score=match_score, print_matches=True, invert=True)
 
-    FileService.df_to_csv(df=french_top_words_per_topic, file_name="topics_fr")
+    # Concat and store in csv
+    best_lsa_matches_by_sentence_transformer = pd.concat([best_lsa_matches_by_sentence_transformer_de_fr, best_lsa_matches_by_sentence_transformer_fr_de])
+    FileService.df_to_csv(df=best_lsa_matches_by_sentence_transformer, file_name="best_lsa_matches_by_sentence_transformer")
 
-    df_topics_fr = french_top_words_per_topic
-    print(df_topics_fr)
+    # Matching by translation
+    lsa_matches_by_translation, lsa_german_counts, lsa_french_counts = TopicMatcherService.match_by_translation(df_lsa_topics_de['Word'], df_lsa_topics_fr['Word'], print_matches=True)
+    FileService.df_to_csv(df=lsa_matches_by_translation, file_name="lsa_matches_by_translation")
```

Comment:
Is there no caching of translations?

Since these are all API calls it makes sense to create a local translation dictionary that can be queried for existing translations. New translations are sent to the API and appended to the dictionary.